### PR TITLE
Migrate Multi-Cloud from dep to Go Modules & Update dependencies for latest Packages

### LINKED ISSUE #939: Migrate Multi-Cloud from dep to Go 1.11+ modules
**Is this a BUG REPORT or FEATURE REQUEST?**:

> Uncomment only one, leave it on its own line: 
>
> /kind feature

**What happened**:  Some packages are unable to update

**What you expected to happen**: Improved Dependency management

**How to reproduce it (as minimally and precisely as possible)**: 

**Anything else we need to know?**:

**Environment**:
- Gelato(release/branch) version:
- OS (e.g. from /etc/os-release):
- Kernel (e.g. `uname -a`):
- Install tools:
- Others:

### PR DESCRIPTION
**Pull request by himanshuvar**
**What this PR does / why we need it**: This is to update the package dependencies to latest.
And to migrate the dep to Go Modules.

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #939 #943 

**Special notes for your reviewer**:

**Release note**:

```release-note
```

### CODE REVIEW

**Comment by kumarashit** [comment]
The changes are for 3000+ files so hard to review. I reviewed go.mod it seems ok.
It's better we do some sanity and update the result here

**Comment by himanshuvar** [comment]
Basic Multi-Cloud Feature Testing is done.

- [ ] 1. Backend Creation
- [ ] 2. Backend Listing
- [ ] 3. Backend Get
- [ ] 4. Backend Delete
- [ ] 5. Bucket Creation
- [ ] 6. Bucket Listing
- [ ] 7. Bucket Get
- [ ] 8. Bucket Delete
- [ ] 9. Object Upload
- [ ] 10. Object Download
- [ ] 11. Object Delete
- [ ] 12. Migration Plan Creation
- [ ] 13. Migration Plan Listing
- [ ] 14. Migration Plan Get
- [ ] 15. Migration Plan Delete
- [ ] 16. Migration Job Execution
- [ ] 17. Migration Job Listing
- [ ] 18. Migration Job Get

**Comment by himanshuvar** [comment]
Migration Job is successful with the following known issues.
These issues are still there after elba. 
https://github.com/sodafoundation/multi-cloud/issues/952
https://github.com/sodafoundation/multi-cloud/issues/940

File: s3/pkg/messagebus/tests/message_bus_suite_test.go [hunk 1]
```diff
@@ -5,8 +5,8 @@ import (
 	"testing"
 
 	"github.com/opensds/multi-cloud/s3/pkg/helper"
-	"github.com/opensds/multi-cloud/s3/pkg/log"
-	bus "github.com/opensds/multi-cloud/s3/pkg/messagebus"
+	log "github.com/sirupsen/logrus"
+        bus "github.com/opensds/multi-cloud/s3/pkg/messagebus"
```
Comment (sfzeng) [inline s3/pkg/messagebus/tests/message_bus_suite_test.go:9]: Plz format this file.

Comment (himanshuvar) [reply s3/pkg/messagebus/tests/message_bus_suite_test.go]: Log references are missing. I am reverting back changes according to development branch.
I have opened an issue for tracking this.
https://github.com/sodafoundation/multi-cloud/issues/975

**Pull request merged**